python/pyspark/streaming/tests.py

```diff
@@ -1013,7 +1009,22 @@ def search_kinesis_asl_assembly_jar():
     kafka_assembly_jar = search_kafka_assembly_jar()
     flume_assembly_jar = search_flume_assembly_jar()
     kinesis_asl_assembly_jar = search_kinesis_asl_assembly_jar()
-    jars = "%s,%s,%s" % (kafka_assembly_jar, flume_assembly_jar, kinesis_asl_assembly_jar)
+    if kinesis_asl_assembly_jar is None:
+        kinesis_jar_present = False
```

Comment:
That is a good point. If that flag is set, then the use definitely wants to run ALL the kinesis tests. In that case its best to fail.